androidApp/src/main/java/com/example/tasky/android/alarm/data/AlarmReceiver.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.tasky.android.alarm.data
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.os.Build
+import com.example.tasky.agenda.domain.model.AgendaType
+
+class AlarmReceiver : BroadcastReceiver() {
+    override fun onReceive(
+        context: Context?,
+        intent: Intent?,
+    ) {
+        val id = intent?.getStringExtra(AlarmIntentKey.ID.name) ?: return
+        val title = intent.getStringExtra(AlarmIntentKey.TITLE.name) ?: return
+        val type =
+            if (Build.VERSION.SDK_INT >= 33) {
+                intent.getSerializableExtra(AlarmIntentKey.TYPE.name, AgendaType::class.java)
+            } else {
+                intent.getSerializableExtra(AlarmIntentKey.TYPE.name) ?: return
+            }
```

Comment:
I had to do multiple of these checks, so I created a extension function for it:
```
inline fun <reified T : Parcelable> Intent.getCompatParcelableExtra(name: String): T? {
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
        getParcelableExtra(name, T::class.java)
    } else {
        getParcelableExtra<T>(name)
    }
}
```

Usage:
```
val data = intent.getCompatParcelableExtra<MyParcelableType>(KEY_DATA)
```